File: app/src/main/java/com/kieran/winnipegbus/Adapters/StopFeaturesAdapter.java
package com.kieran.winnipegbus.Adapters;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.kieran.winnipegbus.R;
import com.kieran.winnipegbusbackend.StopFeature;

import java.util.List;

public class StopFeaturesAdapter extends ArrayAdapter<StopFeature> {
    private int layoutResourceId;
    private List<StopFeature> stopFeatures;
    private LayoutInflater inflater;

    public StopFeaturesAdapter(Context context, int layoutResourceId, List<StopFeature> stopFeatures) {
        super(context, layoutResourceId, stopFeatures);
        this.layoutResourceId = layoutResourceId;
        this.stopFeatures = stopFeatures;
        inflater = ((Activity)context).getLayoutInflater();
    }

    @Override
    public boolean areAllItemsEnabled() {
        return false;
    }

    @Override
    public boolean isEnabled(int position) {
        return false;
    }

    @Override
    public View getView(int position, View row, ViewGroup parent) {
        StopHolder holder;

        if(row == null) {
            row = inflater.inflate(layoutResourceId, parent, false);

            holder = new StopHolder();
            holder.name = (TextView)row.findViewById(R.id.stop_feature_name);
            holder.count = (TextView)row.findViewById(R.id.stop_feature_count);
            row.setTag(holder);
        } else {
            holder = (StopHolder)row.getTag();
        }

        StopFeature stopFeature = stopFeatures.get(position);
        holder.name.setText(stopFeature.getName());
        holder.count.setText(Integer.toString(stopFeature.getCount()));
        return row;
    }

    private static class StopHolder {
        TextView name;
        TextView count;
    }
}

File: app/src/main/java/com/kieran/winnipegbus/Adapters/StopFeaturesAdapter.kt
package com.kieran.winnipegbus.Adapters

import android.app.Activity
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.kieran.winnipegbus.R
import com.kieran.winnipegbusbackend.StopFeature

class StopFeaturesAdapter(context: Context, private val layoutResourceId: Int, private val stopFeatures: List<StopFeature>) : ArrayAdapter<StopFeature>(context, layoutResourceId, stopFeatures) {
    private val inflater: LayoutInflater

    init {
        inflater = (context as Activity).layoutInflater
    }

    override fun areAllItemsEnabled(): Boolean {
        return false
    }

    override fun isEnabled(position: Int): Boolean {
        return false
    }

    override fun getView(position: Int, row: View?, parent: ViewGroup): View {
        var row = row
        val holder: StopHolder

        if (row == null) {
            row = inflater.inflate(layoutResourceId, parent, false)

            holder = StopHolder()
            holder.name = row!!.findViewById<View>(R.id.stop_feature_name) as TextView
            holder.count = row.findViewById<View>(R.id.stop_feature_count) as TextView
            row.tag = holder
        } else {
            holder = row.tag as StopHolder
        }

        val stopFeature = stopFeatures[position]
        holder.name!!.text = stopFeature.name
        holder.count!!.text = Integer.toString(stopFeature.count)
        return row
    }

    private class StopHolder {
        internal var name: TextView? = null
        internal var count: TextView? = null
    }
}
